lib/spack/spack/solver/concretize.lp

```diff
@@ -35,7 +19,7 @@ version_declared(Package, Version, Weight) :- version_declared(Package, Version,
 :- version_declared(Package, Version, Weight, Origin1),
    version_declared(Package, Version, Weight, Origin2),
    Origin1 < Origin2,
-   error("Internal error: two versions with identical weights").
+   internal_error("Two versions with identical weights").
```

Comment:
Not something we need for this PR, but curious if the internal errors could just be optimization criteria, too.  Would it help solve time?

Reply:
They could be optimization criteria. I kept them as constraints because it seemed when I was deciding that switching to optimization criteria was hurting performance, and as long as they were never going to produce an error that meant anything to the user I might as well not hurt performance further.